### PR TITLE
Add Selection Background Color as a setting to Profiles and Col…

### LINKED ISSUE #3326: Add a setting to control selection background color
From @cayos in #3322:

> It'd be amazing if you guys could implement the few remaining color customization options from VSCode's terminal to reach feature parity. They are:
> * terminal.selectionBackground: The selection background color of the terminal.

### PR DESCRIPTION
**Pull request by leonMSFT**
## Summary of the Pull Request
This introduces a setting to both Profiles and ColorSchemes called <code>selectionBackground</code> that allows you to change the selection background color to what's specified. If <code>selectionBackground</code> isn't set in either the profile or color scheme, it'll default to what it was before - white.

## PR Checklist
* [x] Closes #3326
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [x] Tests added/passed
* [x] Requires documentation to be updated

## Validation Steps Performed
- Added selectionBackground to existing profile and colorscheme tests.
- Verified that the color does change to what I expect it to be when I add "selectionBackground" to either/both a profile and a color scheme.

### CODE REVIEW

**Comment by mdtauk** [comment]
Is there already a SelectionForeground colour?

**Comment by leonMSFT** [comment]
@mdtauk Nope! There is not currently a setting for SelectionForeground.

File: src/cascadia/TerminalApp/ColorScheme.cpp [hunk 1]
```diff
@@ -1,222 +1,237 @@
...
-// - returns std::nullopt if `json` doesn't have the `name` property
-std::optional<std::wstring> TerminalApp::ColorScheme::GetNameFromJson(const Json::Value& json)
-{
-    if (const auto name{ json[JsonKey(NameKey)] })
-    {
-        return GetWstringFromJson(name);
-    }
-    return std::nullopt;
-}
+// Copyright (c) Microsoft Corporation.
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/ColorScheme.cpp]: Why does it think this entire file changed?

File: src/cascadia/TerminalApp/ColorScheme.cpp [hunk 2]
```diff
@@ -1,222 +1,237 @@
...
+    "brightCyan",
+    "brightWhite"
+};
+
+ColorScheme::ColorScheme() :
+    _schemeName{ L"" },
+    _table{},
+    _defaultForeground{ RGB(242, 242, 242) },
+    _defaultBackground{ RGB(12, 12, 12) },
+    _selectionBackground{ RGB(242, 242, 242) }
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/ColorScheme.cpp]: Should this (and the others) be `DEFAULT_FOREGROUND` (etc.) from `DefaultSettings.h`?

Comment (leonMSFT) [reply src/cascadia/TerminalApp/ColorScheme.cpp]: Yup, you're right, I've mimicked the default foreground and background from what they're set as in `TerminalSettings.cpp`, and changed selectionBackground to use `DEFAULT_FOREGROUND`.

File: src/cascadia/TerminalApp/ColorScheme.cpp (hunk 1 above)
Comment (leonMSFT) [reply src/cascadia/TerminalApp/ColorScheme.cpp]: some weird whitespace only change 🤷‍♂, i reverted it though.

**Comment by leonMSFT** [comment]
@zadjii-msft  Thanks for the suggestions! Putting the setting in `IControlSettings` and letting `TermControl` update the color in `DxEngine` really cut down on the amount of files I needed to touch.

File: src/renderer/dx/DxRenderer.cpp [hunk 3]
```diff
@@ -41,6 +41,7 @@ DxEngine::DxEngine() :
     _displaySizePixels{ 0 },
     _foregroundColor{ 0 },
     _backgroundColor{ 0 },
+    _selectionBackground{ 0 },
```
Comment (zadjii-msft) [inline src/renderer/dx/DxRenderer.cpp]: Should `_selectionBackground` be `DEFAULT_FOREGROUND` by default?

Note that `OpenConsole.exe` (conhost) can run with the Dx engine as well (by setting `UseDx=1` in the registry under `HKCU\Console`), where it won't have someone setting the selection background.

Comment (mdtauk) [reply src/renderer/dx/DxRenderer.cpp]: If the Selection Background is being set to the Foreground colour by default, there should probably be a Selection Foreground colour, which would be set to the Background colour by default too.

Comment (zadjii-msft) [reply src/renderer/dx/DxRenderer.cpp]: I'm mostly concerned that the default value of this variable has changed from `DEFAULT_FOREGROUND` to `0`. We should probably just file a follow up issue to track that work.

Comment (leonMSFT) [reply src/renderer/dx/DxRenderer.cpp]: Forgot about conhost being able to run with DxEngine! I agree, `_selectionBackground` should be `DEFAULT_FOREGROUND` here.

I'll file an issue to track selection foreground.

**Comment by javierdlg** [comment]
Thanks for the quick turnaround!

**Pull request merged**